sycl/include/CL/sycl/detail/stl_type_traits.hpp

```diff
@@ -50,15 +50,15 @@ template <typename T>
 using iterator_to_const_type_t =
     std::is_const<typename std::remove_pointer<iterator_pointer_t<T>>::type>;
 
-template <class...> using requirements_list = void;
+template <class...> using void_t = void;
```

Comment:
Perhaps you could add a TODO about removing this when we switch to a newer C++ version providing `std::void_t`...
https://en.cppreference.com/w/cpp/types/void_t

Reply:
Almost all the traits in this headers are defining those not yet in C++11 but available in newer versions. I agree we should replace them when requiring a higher C++ minimum version. I don't think we need an extra comment for each of them.

Reply:
Characters are cheap nowadays... Good C++ software engineers guessing some TODO or non existing comments in the future will be quite expensive. :-)

Reply:
Did it for all not just for the single one. Hope no one minds that I moved them to group them based on C++ version.